spec.html

```diff
@@ -40179,6 +40184,7 @@ <h1>`Promise.all` Resolve Element Functions</h1>
             1. Let _index_ be _F_.[[Index]].
             1. Let _values_ be _F_.[[Values]].
             1. Let _promiseCapability_ be _F_.[[Capability]].
+            1. Assert: _promiseCapability_ is a PromiseCapability Record.
```

Comment:
Adding an assertion for only this internal slot seems a bit inconsistent. What makes `F.[[Capability]]` so special that we need to assert its type is a PromiseCapability Record, whereas the types for `F.[[AlreadyCalled]]`, `F.[[Index]]`, `F.[[Values]]`, and `F.[[RemainingElements]]` still need to inferred from the point where `F` was created? If more explicit type descriptions are necessary, I'd suggest to document them as part of the preamble in [25.6.4.1.3 Promise.all Resolve Element Functions](https://tc39.es/ecma262/#sec-promise.all-resolve-element-functions).

Reply:
That's a fair point - I think they *all* need assertions, this PR was only focusing on PromiseCapability Records.

I wouldn't expect, though, the type of an internal slot to be assertable in prose, since the value could change at any point - i'd expect all of them to have assertions at the point they're first extracted.

Reply:
There could be a table with (Name, Type, Description) to describe the internal slots and their usage, if we don't want to put it all in prose. 

And I don't understand "the value could change at any point", why should an internal slot suddenly contain a different type of value? For example for [ordinary objects](https://tc39.es/ecma262/#sec-ordinary-object-internal-methods-and-internal-slots), the spec describes that [[Prototype]] contains either an object or null, and [[Extensible]] contains a boolean value. And that description is deemed to be enough, we don't assert the types whenever we read [[Prototype]] or [[Extensible]].

Reply:
That’s also a fair point. I think we should somehow - ideally without boilerplate everywhere - but given that, for this case at least, I’ll try to move it to prose.